lmfit/models.py

```diff
@@ -352,7 +352,9 @@ def __init__(self, independent_vars=['x'], prefix='', nan_policy='raise',
     def _set_paramhints_prefix(self):
         self.set_param_hint('amplitude', min=0)
         self.set_param_hint('frequency', min=0)
-        self.set_param_hint('shift', min=0, max=tau)
+        bugfix_active = False
+        if not bugfix_active:
+            self.set_param_hint('shift', min=0, max=tau)
```

Comment:
```suggestion
```
I put this here to demonstrate the bug Setting bugfix_active to True this will get tests passing.